tuf/src/metadata.rs

```diff
@@ -710,6 +710,30 @@ impl RootMetadataBuilder {
 
     /// Construct a new `RootMetadata`.
     pub fn build(self) -> Result<RootMetadata> {
+        if self.root_key_ids.is_empty() {
```

Comment:
to avoid duplicating the validation logic, possibly instead make an error variant for failing to create a RoleDefinition while building (or something) that has a MetadataPath field and a source error to hold the RoleDefinition::new error